### PR TITLE
GH945: Fix dotnet restore settings

### PR DESCRIPTION
**Pull request by adamhathcock**
source, fallbacksource and inferedruntimes all allow multiple arguments by specifying the same arg multiple times.

Same fix as this: https://github.com/cake-build/cake/pull/838

### CODE REVIEW

**Comment by adamhathcock** [comment]
Seems like there is a lot wrong?

**Comment by adamhathcock** [comment]
I'm thinking the usage of the dotnet cli isn't correct and some of this needs to be cleaned up.  That would involve breaking changes though.  I can limit what i'm doing to just additive changes but it's less pretty.

File: src/Cake.Common/Tools/DotNetCore/Restore/DotNetCoreRestoreSettings.cs [hunk 1]
```diff
@@ -14,6 +14,11 @@ public sealed class DotNetCoreRestoreSettings : DotNetCoreSettings
         public string Source { get; set; }
```
Comment (devlead) [inline src/Cake.Common/Tools/DotNetCore/Restore/DotNetCoreRestoreSettings.cs]: @adamhathcock please in this PR remove `string Source { get; set; }` and it's usages in favor of the new `ICollection<string> Sources { get; set; }`

We've discussed internally with in @cake-build/team even though it's a breaking change, to keep API clean and consistent, and as we feel we catch this early.

File: src/Cake.Common/Tools/DotNetCore/Build/DotNetCoreBuildSettings.cs [hunk 2]
```diff
@@ -29,6 +29,11 @@ public sealed class DotNetCoreBuildSettings : DotNetCoreSettings
         public string Configuration { get; set; }
```
Comment (devlead) [inline src/Cake.Common/Tools/DotNetCore/Build/DotNetCoreBuildSettings.cs]: @adamhathcock please in this PR remove `string Configuration { get; set; }` and it's usages in favor of the new `ICollection<string> Configurations { get; set; }`

We've discussed internally with in @cake-build/team even though it's a breaking change, to keep API clean and consistent, and as we feel we catch this early.

**Comment by devlead** [comment]
@adamhathcock left some feedback, could you please address, rebase against latest develop, squash into 1 commit and when ping us when you PR is ready for review and we'll look at reviewing asap.

**Comment by adamhathcock** [comment]
After looking at the dotnet cli repo code and some manual testing.  I think I've fixed the issues now. 
Dotnet Restore can have more than one source.  Fallbacksources need to be split out.

When building, you can only have one configuration and one framework at a time.

**Comment by devlead** [comment]
@adamhathcock ah nice catch! Then we'll review this and ping you if we find anything. Meanwhile could you create an issue for this and reference it in this PR.

**Comment by gep13** [comment]
Relates to #945

**Comment by devlead** [comment]
@adamhathcock LGTM :+1: Merged, thanks for contributing!

**Pull request merged**